packages/core/src/modules/credentials/services/RevocationService.ts

```diff
@@ -0,0 +1,114 @@
+import type { V1RevocationNotificationMessage, V2RevocationNotificationMessage } from '../messages'
+
+import { scoped, Lifecycle } from 'tsyringe'
+
+import { AgentConfig } from '../../../agent/AgentConfig'
+import { EventEmitter } from '../../../agent/EventEmitter'
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { CredentialEventTypes } from '../CredentialEvents'
+import { RevocationNotification } from '../models'
+import { CredentialRepository } from '../repository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class RevocationService {
+  private credentialRepository: CredentialRepository
+  private eventEmitter: EventEmitter
+  private logger: Logger
+
+  public constructor(credentialRepository: CredentialRepository, eventEmitter: EventEmitter, agentConfig: AgentConfig) {
+    this.credentialRepository = credentialRepository
+    this.eventEmitter = eventEmitter
+    this.logger = agentConfig.logger
+  }
+
+  private async processRevocationNotification(
+    revocationRegistryId: string,
+    credentialRevocationId: string,
+    connection?: ConnectionRecord,
+    comment?: string
+  ) {
+    const query = { revocationRegistryId, credentialRevocationId }
+    this.logger.trace(`Getting record by query for revocation notification:`, query)
+    const credentialRecord = await this.credentialRepository.getSingleByQuery(query)
+
+    if (!connection) {
+      throw new AriesFrameworkError('No connection record for message')
+    }
+    credentialRecord.assertConnection(connection.id)
+
+    credentialRecord.revocationNotification = new RevocationNotification(comment)
+    await this.credentialRepository.update(credentialRecord)
+
+    this.logger.trace('Emitting RevocationNotificationReceivedEvent')
+    this.eventEmitter.emit<RevocationNotificationReceivedEvent>({
+      type: CredentialEventTypes.RevocationNotificationReceived,
+      payload: {
+        credentialRecord,
+      },
+    })
+  }
+
+  /**
+   * Process a recieved {@link RevocationNotificationMessagev1}. This will create a
+   * {@link RevocationNotification} and store it in the corresponding {@link ConnectionRecord}
+   *
+   * @param messageContext message context of RevocationNotificationMessageV1
+   */
+  public async v1ProcessRevocationNotification(
+    messageContext: InboundMessageContext<V1RevocationNotificationMessage>
+  ): Promise<void> {
+    this.logger.info('Processing revocation notification v1', { message: messageContext.message })
+    // ThreadID = indy::<revocation_registry_id>::<credential_revocation_id>
+    const threadRegex = /(indy)::(.+)::(\d+)$/
+    const threadId = messageContext.message.issueThread
+    try {
+      const threadIdGroups = threadId.match(threadRegex)
+      if (threadIdGroups) {
+        const [, , revocationRegistryId, credentialRevocationId] = threadIdGroups
+        const comment = messageContext.message.comment
+        const connection = messageContext.connection
```

Comment:
If we do `const connection = messageContext.assertReadyConnection` we don't have to make the connection parameter optional in the `processRevocationNotification` method